R/otp-setup.R

```diff
@@ -0,0 +1,290 @@
+#' This function stops an already running OTP instance
+#'
+#' The function assumes you have run otp_setup()
+#'
+#' @export
+otp_stop <- function()
+{
+  if(Sys.info()[['sysname']] == "Linux") {
+    message("You're on linux, this function is not yet supported")
+  }else if(Sys.info()[['sysname']] == "Windows"){
+    readline(prompt="This will force Java to close, Press [enter] to continue, [escape] to abort")
+    system("Taskkill /IM java.exe /F", intern = TRUE)
+  }else if(Sys.info()[['sysname']] == "Darwin"){
+    message("You're on Mac, this function is not yet supported")
+  }else{
+    message("You're on and unknow OS, this function is not yet supported")
+  }
+
+}
+
+
+#' Basic OTP Setup Checks
+#'
+#' Checks to run before setting up the OTP
+#'
+#' @param dir A character string path to a folder containing the necessary files, see details
+#' @param router A character string for the name of the router, must match with contents of dir, default "current"
+#' @param graph Logical, check for graph, default = FALSE
+#'
+otp_checks <- function(otp = NULL, dir = NULL, router = NULL, graph = FALSE)
+{
+  ### Checks
+  # Check we have the directory defined
+  if(!exists("otp")){
+    warning("Path to the Open Trip Planner is not defined")
+    stop()
+  }
+  if(!exists("dir")){
+    warning("Path to the files to build graph are not defined")
+    stop()
+  }
+  # Check that the folder exists
+  if(!dir.exists(otp)){
+    warning(paste0("Unable to find directory: ",otp))
+    stop()
+  }
+  if(!dir.exists(dir)){
+    warning(paste0("Unable to find directory: ",dir))
+    stop()
+  }
+  # Check that the jar exists
+  jar_file <- list.files(otp, recursive = F)
+  jar_file <- jar_file[grepl("\\.jar",jar_file)]
+  if(length(jar_file) == 0){
+    warning(paste0("Unable to find .jar file in ",dir))
+    stop()
+  }
+
+  # Check for the graphs folder
+  if(!dir.exists(paste0(dir,"/graphs"))){
+    warning(paste0("The graphs sub-folder could not be found in: ",dir))
+    stop()
+  }
+  # Check for the router folder
+  if(!dir.exists(paste0(dir,"/graphs/",router))){
+    warning(paste0("The router sub-folder named '",router,"', could not be found in: ",dir,"/graphs"))
+    stop()
+  }
+
+  # Check we have correct verrsion of Java
+  if(Sys.info()[['sysname']] == "Linux") {
+    message("You're on linux, java version check not yet supported")
+  }else if(Sys.info()[['sysname']] == "Windows"){
+    java_version <- try(system("java -version", intern = TRUE))
+    if(class(java_version) == "try-error"){
+      warning("R was unable to detect a version of Java")
+      stop()
+    }else{
+      java_version <- java_version[1]
+      java_version <- strsplit(java_version,"\"")[[1]][2]
+      java_version <- strsplit(java_version,"\\.")[[1]][1:2]
+      java_version <- as.numeric(paste0(java_version[1],".",java_version[2]))
+      if(java_version < 1.8){
+        warning("OTP requires Java version 1.8 or later")
+        stop()
+      }
```

Comment:
@mem48

AFAIK JAVA version must be 1.8. It won't work on 9 or 10.

Reply:
According to http://docs.opentripplanner.org/en/latest/Basic-Tutorial/#start-up-otp it is 1.8 or higher. I'm cautious of being too prescriptive as it will mean updating the package each time OTP or Java get an update. It was more intended as a test that the R to Java communication was working.

Reply:
That tutorial is incorrect. See issue raised by Robin. Get your point about being too prescriptive but OTP is likely to be very slow in updating for new Java versions. Maybe more important in the Vignettes that it is clear otherwise likely to cause huge frustration.

Reply:
Ok, changed to check that version is 8.X